typedargs/doc_parser.py

```diff
@@ -355,25 +358,29 @@ def parse_return(return_line, include_desc=False):
 
     ret_def, _colon, desc = return_line.partition(':')
     if _colon == "":
-        raise ValidationError("Invalid return declaration in docstring, missing colon", declaration=ret_def)
+        if validate_type:
+            raise ValidationError("Invalid return declaration in docstring, missing colon", declaration=ret_def)
+
+        desc = ret_def
+        ret_def = None
 
     if not include_desc:
         desc = None
 
-    if 'show-as' in ret_def:
+    if ret_def and 'show-as' in ret_def:
```

Comment:
I'm not sure this change is needed.

`ref_def` comes from `partition` which guarantees that it will be a `str`, possibly `""`.  So this new line is logically equivalent to the original `'show-as' in ret_def` since that evaluates to false without error when given the empty string as input.

Reply:
@timburke 
`ref_def` could be None because of reassigning on line 365.

Function we are discussing is here: https://github.com/iotile/typedargs/blob/support-function-annotations/typedargs/doc_parser.py#L359-L386

Passed `return_line` argument could be without `:`. I have made it to support case when the return value type is specified in annotations and docstring `Returns` stanza has only a description.
In this case this reassigning happens on lines 364-365:
```
        desc = ret_def
        ret_def = None
```
This function returns `return ReturnInfo(None, ret_def, None, True, desc)` where `ret_def` should be `None` or a type name. That is why I decided to assign `ret_def = None` there.

I can change line 365  to `ret_def = ""` and get rid of first condition in `if ret_def and 'show-as' in ret_def:`. But in this case would need to add 2 more lines
```
if ret_def == "":
    ret_def = None
```
If it looks better for you I would change, just let me know.

Reply:
No that's fine as is, I just misread the code.  Thanks for the clarification.